yarn-project/archiver/src/archiver/archiver.ts

```diff
@@ -538,6 +547,8 @@ export class Archiver extends EventEmitter implements ArchiveSource, Traceable {
           globalVariables: block.block.header.globalVariables.toInspect(),
         });
       }
+      // Set the synced L1 block number here as that containing the most recently retrieved L2 block
+      await this.store.setBlockSynchedL1BlockNumber(lastProcessedL1BlockNumber);
```

Comment:
This should not be needed, since the store automatically sets this value upon calling `addBlocks`

https://github.com/AztecProtocol/aztec-packages/blob/ccac9ff3cf460822445e9c35fc2c40b4f142035a/yarn-project/archiver/src/archiver/kv_archiver_store/block_store.ts#L88

I'm fine moving the check here to be more explicit, but if we do, we need to delete the one from the block-store, and check it was not needed from somewhere else.

Reply:
Oh that's confusing! Thanks for picking that up. I searched for all calls to `setSynchedL1BlockNumber` without realising the store was making up it's own mind about when to set the block number.

Reply:
Yeah, I had already spent quite some time figuring out how that checkpoint was being updated, to find it happened hidden inside the blockstore. It may be a good idea to remove it from there.